tools/challenge-helper-scripts/create-quiz.ts

```diff
@@ -0,0 +1,206 @@
+  };
+  void withTrace(
+    fs.writeFile,
+    introJsonPath,
+    await format(JSON.stringify(newIntro), { parser: 'json' })
+  );
+}
+
+async function createMetaJson(
+  superBlock: SuperBlocks,
+  block: string,
+  title: string,
+  helpCategory: string,
+  challengeId: ObjectID
+) {
+  const metaDir = path.resolve(__dirname, '../../curriculum/challenges/_meta');
+  const newMeta = await parseJson<Meta>('./quiz-meta.json');
+  newMeta.name = title;
+  newMeta.dashedName = block;
+  newMeta.helpCategory = helpCategory;
+  newMeta.superBlock = superBlock;
+  // eslint-disable-next-line @typescript-eslint/no-base-to-string
+  newMeta.challengeOrder = [{ id: challengeId.toString(), title: title }];
+  const newMetaDir = path.resolve(metaDir, block);
+  if (!existsSync(newMetaDir)) {
+    await withTrace(fs.mkdir, newMetaDir);
+  }
+
+  void withTrace(
+    fs.writeFile,
+    path.resolve(metaDir, `${block}/meta.json`),
+    await format(JSON.stringify(newMeta), { parser: 'json' })
+  );
+}
+
+async function createIntroMD(superBlock: string, block: string, title: string) {
+  const introMD = `---
+title: Introduction to the ${title}
+block: ${block}
+superBlock: ${superBlock}
+---
+
+## Introduction to the ${title}
+
+This page is for the ${title}
+`;
+  const dirPath = path.resolve(
+    __dirname,
+    `../../client/src/pages/learn/${superBlock}/${block}/`
+  );
+  const filePath = path.resolve(dirPath, 'index.md');
+  if (!existsSync(dirPath)) {
+    await withTrace(fs.mkdir, dirPath);
+  }
+  void withTrace(fs.writeFile, filePath, introMD, { encoding: 'utf8' });
+}
+
+async function createQuizChallenge(
+  superBlock: SuperBlocks,
+  block: string,
+  title: string
+): Promise<ObjectID> {
+  const superBlockSubPath = getSuperBlockSubPath(superBlock);
+  const newChallengeDir = path.resolve(
+    __dirname,
+    `../../curriculum/challenges/english/${superBlockSubPath}/${block}`
+  );
+  if (!existsSync(newChallengeDir)) {
+    await withTrace(fs.mkdir, newChallengeDir);
+  }
+  return createQuizFile({
+    challengeType: 8,
+    projectPath: newChallengeDir + '/',
+    title: title,
+    dashedName: block
+  });
+}
+
+function parseJson<JsonSchema>(filePath: string) {
+  return withTrace(fs.readFile, filePath, 'utf8').then(
+    // unfortunately, withTrace does not correctly infer that the third argument
+    // is a string, so it uses the (path, options?) overload and we have to cast
+    // result to string.
+    result => JSON.parse(result as string) as JsonSchema
+  );
+}
+
+// fs Promise functions return errors, but no stack trace.  This adds back in
+// the stack trace.
+function withTrace<Args extends unknown[], Result>(
+  fn: (...x: Args) => Promise<Result>,
+  ...args: Args
+): Promise<Result> {
+  return fn(...args).catch((reason: Error) => {
+    throw Error(reason.message);
+  });
+}
+
+void prompt([
+  {
+    name: 'superBlock',
+    message: 'Which certification does this belong to?',
+    default: SuperBlocks.RespWebDesign,
```

Comment:
the old responsive web design cert doesn't have quizzes. So a better default would be the full stack cert